src/bootstrap/bootstrap.py

```diff
@@ -149,25 +150,43 @@ def run(args, verbose=False, exception=False, **kwargs):
 
 
 def stage0_data(rust_root):
+    """Build a dictionary from stage0.txt"""
     nightlies = os.path.join(rust_root, "src/stage0.txt")
-    data = {}
     with open(nightlies, 'r') as nightlies:
-        for line in nightlies:
-            line = line.rstrip()  # Strip newline character, '\n'
-            if line.startswith("#") or line == '':
-                continue
-            a, b = line.split(": ", 1)
-            data[a] = b
-    return data
+        lines = [line.rstrip() for line in nightlies
+                 if not line.startswith("#")]
+        return dict([line.split(": ", 1) for line in lines if line])
 
 
 def format_build_time(duration):
+    """Return a nicer format for build time
+
+    >>> format_build_time('300')
+    '0:05:00'
+    """
     return str(datetime.timedelta(seconds=int(duration)))
 
 
 class RustBuild(object):
+    """Provide all the methods required to build Rust"""
+    def __init__(self):
+        self.cargo_channel = ''
+        self.date = ''
+        self._download_url = 'https://static.rust-lang.org'
+        self.rustc_channel = ''
+        self.build = ''
+        self.build_dir = os.path.join(os.getcwd(), "build")
+        self.clean = False
+        self.config_mk = ''
+        self.config_toml = ''
+        self.printed = False
+        self.rust_root = os.path.abspath(os.path.join(__file__, '../../..'))
+        self.use_locked_deps = ''
+        self.use_vendored_sources = ''
+        self.verbose = False
 
     def download_stage0(self):
+        """Download stage0"""
```

Comment:
@alexcrichton Thanks for the review! I agree with you, I added more information in both methods as you suggested. Please let me know what do you think.